src/puppetlabs/puppetdb/http/catalogs.clj

```diff
@@ -48,10 +48,12 @@
                       (str (:url-prefix globals))))
 
     [node "edges" &]
-    (comp (edges/edges-app version false) (partial http-q/restrict-query-to-node node))
+    (-> (comp (edges/edges-app version false) (partial http-q/restrict-query-to-node node))
+        (wrap-with-parent-check version :node node))
 
     [node "resources" &]
-    (comp (resources/resources-app version false) (partial http-q/restrict-query-to-node node))))
+    (-> (comp (resources/resources-app version false) (partial http-q/restrict-query-to-node node))
+        (wrap-with-parent-check version :node node))))
 
 (defn catalog-app
   [version]
```

Comment:
Well, we should be beholden to the HTTP side of things, if the parent folder returns a 404, so do children so yes this is inconsistent. Whether this is the correct behaviour generally thats a different story, I guess I feel it is correct, in the sense that the key for a catalog is the certname today, and that catalog doesn't exist - regardless of the node state.